tools/htdocs/components/20_VRForm.js

```diff
@@ -0,0 +1,447 @@
+        }
+        if (cons['sift_prediction']) {
+          details.push('<b>SIFT:</b> ' + cons.sift_prediction);
+        }
+        if (cons['polyphen_prediction']) {
+          details.push('<b>PolyPhen:</b> ' + cons.polyphen_prediction);
+        }
+        if (cons['distance']) {
+          details.push('<b>Distance to transcript:</b> ' + cons.distance + 'bp');
+        }
+        if (!details.length) {
+          details = ['-'];
+        }
+
+        cols.push($.map(details, function(detail) { return '<p class="no-bottom-margin">' + detail + '</p>' }));
+
+        return [ cols ];
+      }));
+    }
+
+    // add table, but only if there is data in it
+    if (tableRows.length) {
+      this.elLk.previewDiv.find('._preview_table').append(generateTable(['Gene/Feature/Type', 'Consequence', 'Details'], tableRows));
+    }
+
+    // add info telling user this is not the full result set
+    this.elLk.previewDiv
+      .find('a.zmenu').on('click', function(e) {
+        e.preventDefault();
+        Ensembl.EventManager.trigger('makeZMenu', this.innerHTML.replace(/\W/g, '_'), { event: e, area: {link: $(this)}});
+      }).end()
+      .find('._ht').helptip().end()
+      .find('._close_button').on('click', function() {
+        panel.elLk.previewDiv.empty();
+        panel.enablePreviewButton();
+      });
+
+    // fix preview div's position and dimensions
+    this.elLk.previewDiv.addClass('active');
+  },
+
+  previewError: function(message) {
+  /*
+   * Show error regarding the vep preview in an alert box
+   */
+    this.elLk.previewDiv.removeClass('active loading');
+    this.enablePreviewButton();
+    alert("Unable to generate preview:\n" + message);
+  },
+
+  enablePreviewButton: function(flag) {
+  /*
+   * Enable/disable preview button
+   */
+    if (flag === false) {
+      this.elLk.previewButton.addClass('disabled').prop('disabled', true);
+    } else {
+      this.elLk.previewButton.removeClass('disabled').removeAttr('disabled');
+    }
+  },
+
+  populateForm: function(jobsData) {
+  /*
+   * Populates the form according to the provided ticket data
+   */
+    if (jobsData && jobsData.length) {
+      jobsData = $.extend({}, jobsData[0].config, jobsData[0]);
+      this.base([jobsData]);
+
+      if (jobsData['input_file_type']) {
+        this.elLk.form.find('input[name=file]').parent().append('<p class="_download_link">' + ( jobsData['input_file_type'] === 'text'
+          ? 'Click <a href="' + jobsData['input_file_url'] + '">here</a> to download the previously uploaded file.'
+          : 'You previously uploaded a compressed file to run this job.'
+        ) + '</p>');
+      }
+
+      if (!this.elLk.speciesDropdown.find('input:checked').length) {
+        this.elLk.speciesDropdown.find('input[type=radio]').first().click();
+      }
+
+      this.resetSpecies(jobsData['species']);
+
+      this.elLk.dataField.trigger('change');
+    }
+  },
+
+  reset: function() {
```

Comment:
Need to be reviewed since there are some VEP functionality in there.

Reply:
I tested VR without this function and there was no difference. I'm going to comment this part.